#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <string>
#include <QFileDialog>
#include <QTextStream>
#include <QTableWidgetItem>
#include <chrono>

QT_BEGIN_NAMESPACE
namespace Ui { class MainWindow; }
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

private slots:
    void on_count_valueChanged(int arg1);

    void on_pushButton_2_clicked();

    void on_pushButton_3_clicked();

    void on_pushButton_clicked();

    void on_pushButton_4_clicked();

    void on_pushButton_5_clicked();

private:
    Ui::MainWindow *ui;
    QString *arr;
    QString *changed_arr_1;
    int size;
    int changed_size;
    QString sities [238] = {"Novo Aleksandrovka",
                "Zelenohirs’ke",
                "Beshtarym",
                "Sofiental",
                "Zazerki",
                "Sharpanchi",
                "Serpneve",
                "Trestsyanets",
                "Vilkovo",
                "Karow",
                "Bylasovka",
                "Sovkhoz Osipenko",
                "Khutor Troyany",
                "Fere-Champenoise",
                "Pereyaslivka",
                "Corohod",
                "Hryzivka",
                "Homenky",
                "Chornohorodka",
                "Rostyazhin",
                "Stalineshti",
                "Pyrogyv",
                "Blidcha",
                "Ananjew",
                "Krasuv",
                "Onut",
                "Bohdan",
                "Malaya Berezovitsa",
                "Khitar",
                "Rybno",
                "Horodnychza",
                "Poloshky",
                "Dzvinyacha",
                "Vuynyshce",
                "Vilovka",
                "Vershina-Kamenka",
                "Pavdovo",
                "Osykovo",
                "Voloshinovo",
                "Prostoroye",
                "Blagodatny",
                "Belo-Kamenka",
                "Letychiv",
                "Mironovskiy",
                "Kotsmazov",
                "Prychil",
                "Solonka Velikaya",
                "Malashevka",
                "Schastivaya",
                "Volchytsk",
                "Tsyganskiy Khutor",
                "Dnestshik Denbovy",
                "Nemoviche",
                "Krayevshchina",
                "Gostuv",
                "Khmelevaya",
                "Malaya Shishovka",
                "Mishurin Rog",
                "Bulavinskoye",
                "Koloniya Ist",
                "Zernove",
                "Stavky",
                "Mykhaylivka-Rubezhivka",
                "Grabich",
                "Zakruzhtsy",
                "Kivshyk",
                "Obryv",
                "Fridrovtsy",
                "Kopachi",
                "Garyn’grud",
                "Velikiye Puzyr’ki",
                "Dobrini",
                "Ivan’ki",
                "Byancky",
                "Velikiy Rozhin",
                "Pishchiki",
                "Korolenka",
                "Khutora Solontsy",
                "Novaya Step’",
                "Zabuchchya",
                "Kulichkov",
                "Pukasovtsy",
                "Fedotove",
                "Holovy",
                "Filonove",
                "Dement’yevo",
                "Duninuv",
                "Carpenkove",
                "Madziarki",
                "Ruvnya",
                "Gasenki",
                "Cychury",
                "Malyye Yerchiki",
                "Ovrazhe",
                "Skripitsa",
                "Verguny-Pologi",
                "Skulyn",
                "Solotyne",
                "Sorochanovo",
                "Tutovoye",
                "Gradki",
                "Beres’k",
                "Berezhnitsya Persha",
                "Shirochany",
                "Lopukhiv",
                "Borbin",
                "Makhnivka",
                "Novo-Mayorskoye",
                "Lebid",
                "Alsu",
                "Krychove",
                "Ostrov",
                "Zalissya",
                "Chervonodvorka",
                "Kryuchik",
                "Loshkovtsy",
                "Vykhopki",
                "Serechowicze",
                "Medvedenki",
                "Zaderyivka",
                "Uritskoye Nomer Shestnadtsat’",
                "Velikiy Karabcheyev",
                "Pomirky",
                "Kukhari",
                "Simonenko",
                "Kuz’mino",
                "Stshalkovtse",
                "Pliskachevka",
                "Nemyl’nya",
                "Korobchino",
                "Chymakove",
                "Vasil’kovtsy",
                "Venslavy",
                "Lyubovichi",
                "Boratyn",
                "Korytnytsya",
                "Blystavytsya",
                "Chernecha",
                "Cherneve",
                "Vengershchizna",
                "Karpovo-Krepinskoye",
                "Reshetilovka",
                "Mariampol’",
                "Franopol’",
                "Yasna Polyana",
                "Glazovoye",
                "Subottsi",
                "Makiyevka",
                "Khutor Degtyary",
                "Boluyarka",
                "Novoarkhangel’sk",
                "Norycza",
                "Teotipol’",
                "Kholoduvka",
                "Gogolev",
                "Svyshcyv",
                "Abuzlar",
                "Bilovyshneve",
                "Sen’kov",
                "Rozrovichy",
                "Zadunayevka",
                "Dihtiv",
                "Shpakovo",
                "Staryy Nizhborok",
                "Polyana",
                "Yasna Zirka",
                "Dorobratovo",
                "Probuzhdeniye",
                "Grozov",
                "Stary Burnak",
                "Rekliniec",
                "Malaya Volitsa",
                "Severnyy",
                "Podlesny Mukarov",
                "Malo-Solonoye",
                "Podgornaya",
                "Dyvyn",
                "Kolosovskiye",
                "Shyshkyne",
                "Gnilichki",
                "Khutor Volchanskiy",
                "Yenatskiy",
                "Oleksandrivka",
                "Shcheploty",
                "Zamlyne",
                "Dzvinyache",
                "Al’binovka",
                "Rohizky",
                "Kominy",
                "Werbcze Wielkie",
                "Konovalove",
                "Zelen’",
                "Mokra Kalyhirka",
                "Trakhtemirov",
                "Ugnev",
                "Golaya",
                "Narbutivka",
                "Dzelentsy",
                "Velikaya Gorozhanka",
                "Mozoleyevka",
                "Neplyuyeve",
                "Rotmistrivka",
                "Stozhek",
                "Dvadtsatiletiye Oktyabr’",
                "Kasatin",
                "Grudek Yagellon’ski",
                "Silizovka",
                "Nove-Misto",
                "Kutin",
                "Sladkovodnoye",
                "Slavechno",
                "Bayrach",
                "Buromka",
                "Shevchenkivka",
                "Svobodne",
                "Visichi",
                "Khlopkov",
                "Dzerzhanovka",
                "Krasnollov",
                "Busha",
                "Novorybatcke",
                "Vorob’yлvka",
                "Yevgenivka",
                "Gin’kovtse",
                "Sitayevka",
                "Chervona Stinka",
                "Horoduni",
                "Novospas’ke",
                "Novoselitsa Vtoraya",
                "Kurhans’ke",
                "Verkhnedorozhnoye",
                "Cuchi",
                "Shaterniki",
                "Gorskoye",
                "Greko-Timofeyevskiy",
                "Myakoty",
                "Nadchitsy",
                "Borshchevka"};
};
#endif // MAINWINDOW_H
